metrics/scaling/k8s_scale_net.sh

```diff
@@ -0,0 +1,288 @@
+
+	info "launch [$launch_time_ms]"
+
+	metrics_json_close_array_element
+}
+
+init() {
+        info "Initialising"
+
+	local cmds=("bc" "jq")
+	check_cmds "${cmds[@]}"
+
+	info "Checking Kubernetes accessible"
+	local worked=$( kubectl get nodes > /dev/null 2>&1 && echo $? || echo $? )
+	if [ "$worked" != 0 ]; then
+		die "kubectl failed to get nodes"
+	fi
+
+	info $(get_num_nodes) "Kubernetes nodes in 'Ready' state found"
+
+	k8s_api_init
+
+	 # And now we can set up our results storage then...
+	 metrics_json_init "k8s"
+	 save_config
+}
+
+save_config() {
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"testname": "${TEST_NAME}",
+                "NUM_PODS": ${NUM_PODS},
+                "STEP": ${STEP},
+                "wait_time": ${wait_time},
+                "delete_wait_time": ${delete_wait_time},
+                "settle_time": ${settle_time}
+        }
+EOF
+)"
+        metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+run() {
+	info "Running test"
+	local header_post="Content-Type: application/json"
+	local header_patch="Content-Type: application/strategic-merge-patch+json"
+	local base_curl=${API_ADDRESS}:${API_PORT}/apis/apps/v1/namespaces/default/deployments
+
+	trap cleanup EXIT QUIT KILL
+
+	metrics_json_start_array
+
+	for reqs in $(seq ${STEP} ${STEP} ${NUM_PODS}); do
+		info "Testing replicas ${reqs} of ${NUM_PODS}"
+		# Generate the next yaml file
+
+		local runtime_command
+		if [ -n "$RUNTIME" ]; then
+			runtime_command="s|@RUNTIMECLASS@|${RUNTIME}|g"
+		else
+			runtime_command="/@RUNTIMECLASS@/d"
+		fi
+
+		local input_template
+		local generated_file
+		if [ "$use_api" != "no" ]; then
+			input_template=$input_json
+			generated_file=$generated_json
+		else
+			input_template=$input_yaml
+			generated_file=$generated_yaml
+		fi
+
+		sed -e "s|@REPLICAS@|${reqs}|g" \
+			-e $runtime_command \
+			-e "s|@DEPLOYMENT@|${deployment}|g" \
+			-e "s|@LABEL@|${LABEL}|g" \
+			-e "s|@LABELVALUE@|${LABELVALUE}|g" \
+			-e "s|@GRACE@|${grace}|g" \
+			< ${input_template} > ${generated_file}
+
+		info "Applying changes"
+		local start_time=$(date +%s%N | cut -b1-13)
+
+		if [ "$use_api" != "no" ]; then
+			# If this is the first launch of the deploy, we need to use a different URL form.
+			if [ $reqs == ${STEP} ]; then
+				curl -s ${base_curl} -XPOST -H "${header_post}" -d@${generated_file} > /dev/null
+			else
+				curl -s ${base_curl}/${deployment} -XPATCH -H "${header_patch}" -d@${generated_file} > /dev/null
+			fi
+		else
+			kubectl apply -f ${generated_file}
+		fi
+
+		kubectl rollout status --timeout=${wait_time}s deployment/${deployment}
+		kubectl expose deployment $deployment
+		if [ $? -ne 0 ]; then
+			echo "kubectl expose failed"
+			exit
+		fi
+
+		while [ $? -ne 0 ]; do
+			kubectl get services $deployment -n default --no-headers=true
+		done
+
+		IP=$(kubectl get services $deployment -n default --no-headers=true | awk '{printf $3}')
```

Comment:
I know you said in the PR that this is WIP, and that it currently only works for a single IP/pod :-) - what is the plan though - can we discuss and detail what we intend to measure - is it the single response time of the 'deployment', or are we going to measure the response time of each pod? I'm not even sure what/how they are possible - suspect we should pull @dklyle in for some thoughts and input as well...

Reply:
The replicas for each deployment will be one now, so we'll get only one IP, so we'll measure a single response time of such deployment.